src/children.js

```diff
@@ -3,21 +3,43 @@
  * @param {object | object[]} node or nodes
  * @returns {object[]}
  */
-export default function children (node) {
+export function of (node) {
+	return getPaths(node).map(({node}) => node);
+}
+
+/**
+ * Get a node's children and the corresponding properties and indices
+ * @param {object | object[]} node or nodes
+ * @returns {object[]}
+ */
+export function getPaths (node) {
 	if (Array.isArray(node)) {
-		return node.flatMap(node => children(node));
+		// when node is an array, flatten to avoid nested arrays of children
+		return node.flatMap(node => getPaths(node));
 	}
 
-	let nodeProperties = properties[node.type] ?? [];
-	return nodeProperties.flatMap(property => node[property] ?? []);
+	const childProperties = properties[node.type] ?? [];
+	let children = [];
+
+	for (const property of childProperties) {
+		const child = node[property];
+		// When the node is an array, we want to include the index in the result
+		if (Array.isArray(child)) {
+			children = children.concat(child.map((c, index) => ({node: c, property, index})));
```

Comment:
Nit, but I'd find this clearer to read:
```suggestion
			children.push(child.map((c, index) => ({node: c, property, index})));
```

Reply:
If we did it this way we'd need to add the spread operator for the push, would that add performance downside?

Reply:
I looked it up and I think it would unfortunately